core/include/keyman/keyman_core_api.h

```diff
@@ -1140,6 +1140,55 @@ km_core_keyboard_load(km_core_path_name kb_path,
 
 -------------------------------------------------------------------------------
 
+# km_core_keyboard_load_from_blob()
+
+## Description
+
+Parse and load keyboard from the supplied blob and a pointer to the loaded keyboard
+into the out paramter.
+
+## Specification
+
+```c */
+KMN_API
+km_core_status km_core_keyboard_load_from_blob(void* blob, size_t blob_size,
+                                               km_core_keyboard** keyboard);
+
+/*
+```
+
+## Parameters
+
+`blob`
+: a byte array containing the content of a KMX/KMX+ file
+
+`blob_size`
+: A pointer to a size_t variable with the size of the blob in bytes.
+
+`keyboard`
+: A pointer to result variable: A pointer to the opaque keyboard
+  object returned by the Processor. This memory must be freed with a
+  call to [km_core_keyboard_dispose].
+
+## Returns
+
+`KM_CORE_STATUS_OK`
+: On success.
+
+`KM_CORE_STATUS_NO_MEM`
+: In the event an internal memory allocation fails.
+
+`KM_CORE_STATUS_IO_ERROR`
+: In the event the keyboard file is unparseable for any reason
```

Comment:
This is a bit of a funny code but matches km_core_keyboard_load I guess

Reply:
Slightly changed the wording.